CONTRIBUTING.md

```diff
@@ -0,0 +1,72 @@
+# Contributing
+Oracle welcomes contributions to this repository from anyone.
+
+If you want to submit a pull request to fix an bug or enhance an existing
+`Dockerfile`, please first open an issue and link to that issue when you
+submit your pull request.
+
+If you have any questions about a possible submission, feel free to open
+an issue too.
+
+## Pull Request Process
+1. Fork this repository
+2. Create a branch in your fork to implement the changes. We recommend using
+the issue number as part of your branch name, e.g. `1234-fixes`
+3. Ensure that any documentation is updated with the changes that are required
+by your fix.
+4. Submit the pull request. *Do not leave the pull request blank*. Explain exactly
+what your changes are meant to do and provide simple steps on how to validate
+your changes. We will assign the pull request to 2-3 people for review before
+it is merged.
+
+## Golden Rules
+
+We have some golden rules that we require all submitted `Dockerfiles` to abide
+by. These rules are provided by Oracle Global Product Security and may change
+at any time.
+
+Most of these are targeted at Oracle employees, but apply to anyone who submits
+a pull request.
+
+### Base Image Rules
+
+1. All Oracle product images must use an Oracle Linux base image.
+1. Extend an existing product image wherever possible. For example, if your
+product requires WebLogic, then extend the WebLogic image instead of creating
+your own WebLogic installation.
+1. Re-use existing scripts wherever possible. If a particular base image or
+script doesn't have the functionality you need, open an issue and work with
+the image owner to implement it.
+1. Specify a fixed version in the `FROM` directive, i.e. use
+`FROM oraclelinux:7-slim` or `FROM java/serverjre:8`.
+
+### Security-related Rules
+
+1. Do not require the use of the `--privileged` flag when running a container.
+
+1. Do not run `ssh` inside a container.
+1. Do not use host networking mode (`--net=host`) for a container.
+1. Do not hard-code any passwords.
```

Comment:
This should probably elaborate on that it's best practice to auto generate a password by default via openssl rand and accept a password as argument.